### PR TITLE
Adds in basic collection crud

### PR DESCRIPTION
**Pull request by mejackreed**
Starts to address #19

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+##
+# Controller used for a user's collection
+class CollectionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_collection, only: [:show, :edit, :update, :destroy]
+  load_and_authorize_resource
+
+  # GET /collections
+  def index
+    @collections = Collection.where(user: current_user)
```
Comment (jkeck) [inline app/controllers/collections_controller.rb]: Can we use `accessible_by` here?  I can't quite remember if that's the use-case for this or not.

File: app/controllers/collections_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  def edit
+  end
+
+  # POST /collections
+  def create
+    @collection = Collection.new(collection_params)
+    @collection.user = current_user
+
+    if @collection.save
+      redirect_to @collection, notice: 'Collection was successfully created.'
```
Comment (jkeck) [inline app/controllers/collections_controller.rb]: Should we i18n these strings?

File: app/models/ability.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    ##
+    # A user can only read or manage their collections
+    can :read, Collection, user_id: user.id
```
Comment (jkeck) [inline app/models/ability.rb]: Is this redundant? I think `manage` should grant `read` access.

File: app/views/collections/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+      <ul>
+      <% collection.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+  
+  <div class="form-group row">
```
Comment (jkeck) [inline app/views/collections/_form.html.erb]: `bootstrap_form` could allow us to dry up this form markup a bit.  Any thoughts on pulling that in?

File: spec/controllers/collections_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  end
+
+  describe 'PUT #update' do
+    context 'with valid params' do
+      let(:new_attributes) { { name: 'holla' } }
+
+      it 'updates the requested collection' do
+        collection = Collection.create! valid_attributes
+        put :update, params: {id: collection.to_param, collection: new_attributes}
+        collection.reload
```
Comment (jkeck) [inline spec/controllers/collections_controller_spec.rb]: Is this doing anything for us in this test?

File: app/views/collections/_form.html.erb (hunk 4 above)
Comment (mejackreed) [inline app/views/collections/_form.html.erb]: I setup project to use Bootstrap 4 (why start a new project on an old version). AFAICT there wasn't support for 4 yet.

Comment (jkeck) [inline app/views/collections/_form.html.erb]: It appears they have a branch for support for BS4.  That being said, now that I think about this more, we're probably going to customize these forms significantly enough that something like `bootstrap_form` might not end up having that much payoff in the end.

**Pull request merged**